frontend/src/composables/useCloudProfile/useVolumeTypes.js

```diff
@@ -0,0 +1,114 @@
+//
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import {
+  computed,
+  isRef,
+} from 'vue'
+
+import { useRegions } from './useRegions.js'
+
+import get from 'lodash/get'
+import filter from 'lodash/filter'
+import includes from 'lodash/includes'
+import find from 'lodash/find'
+import map from 'lodash/map'
+import intersection from 'lodash/intersection'
+
+/**
+ * Composable for cloud profile volume type management
+ * @param {Ref<CloudProfile>} cloudProfile - Vue ref containing the cloud profile object
+ * @returns {Object} Object containing computed properties and functions for volume types
+ */
+export function useVolumeTypes (cloudProfile) {
+  if (!isRef(cloudProfile)) {
+    throw new Error('cloudProfile must be a ref!')
+  }
+
+  const { useZonesByRegion } = useRegions(cloudProfile)
+
+  /**
+   * Get all volume types from cloud profile
+   */
+  const volumeTypes = computed(() => {
+    return get(cloudProfile.value, ['spec', 'volumeTypes'], [])
+  })
+
+  /**
+   * Get volume types filtered by region availability
+   * @param {Ref<String>} region - Vue ref containing the region name
+   * @returns {ComputedRef<Array>} Computed ref of available volume types
+   */
+  function useVolumeTypesByRegion (region) {
+    if (!isRef(region)) {
+      throw new Error('region must be a ref!')
+    }
+
+    if (!cloudProfile.value) {
+      return computed(() => [])
+    }
```

Comment:
this is not correct and would break reactivity. the unset cloudProfile.value check has to be done within the computed below